metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -2145,13 +2151,20 @@ class MetalsLspService(
               .error("Error re-importing Scala CLI build", exception)
         }
 
-    if (otherChanges0.nonEmpty)
-      quickConnectToBuildServer().onComplete {
-        case Failure(e) =>
-          scribe.warn("Error refreshing build", e)
-        case Success(_) =>
-          scribe.info("Refreshed build after change")
+    if (otherChanges0.nonEmpty) {
+      bspSession match {
+        case None => scribe.warn("No build server connected")
+        case Some(session) =>
+          for {
+            _ <- importBuild(session)
+            _ <- indexer.profiledIndexWorkspace(runDoctorCheck)
+          } {
+            focusedDocument().foreach(path => compilations.compileFile(path))
+            compilers.cancel()
```

Comment:
`importBuild` does that too, why do we need this again?

Reply:
Hmm... it wasn't working without it, I think we might actually need to move it as last.